Wrote Feature that allows us to switch the networkpath

amitsandhel opened the pull request:
We can open a new console and model on defined pre-existing network file. The python side does extract the path and opens a new console. Only issue is that resulting calls to run a second time. The pipe is open though.
Also moved the opening of the console and model into its own function called loadModel making it more flexible and robust.
Some grammar typos will need to be fixed as well

amitsandhel committed:
Wrote Feature that allows us to switch the networkpath
We can open a new console and model on defined pre-existing network file. The python side does extract the path and opens a new console. Only issue is that resulting calls to run a second time. The pipe is open though.
Also moved the opening of the console and model into its own function called loadModel making it more flexible and robust

JamesVaughan commented on src/TMG.Aimsun/AimsunController.cs:
```diff
@@ -216,6 +221,38 @@ private void EnsureWriteAvailable(IModule caller)
                 throw new XTMFRuntimeException(caller, "Aimsun Bridge was invoked even though it has already been disposed.");
             }
         }
+        public bool SwitchModel(IModule caller, string macroName, string jsonParameters)
```
Why is the parameter called macroName when you it should be the file path to the Aimsun project, and why are we taking in JSON parameters?

JamesVaughan commented on aimsunBridge.py:
```diff
@@ -190,21 +192,41 @@ def sendSuccess(self):
+            # extract the name of the tool along with the parameters and pass it to the function
+            self.NetworkPath = self.readString()
+            print (self.NetworkPath)
+            console, model = self.loadModel()
```
We don't want to create a new instance of the console, only invoke the open method and get the new model object.

JamesVaughan commented on src/TMG.Aimsun/AimsunController.cs:
```diff
@@ -216,6 +221,38 @@ private void EnsureWriteAvailable(IModule caller)
+                        {
+                            writer.Write(jsonParameters.Length);
+                            writer.Write(jsonParameters.ToCharArray());
+                        }
```
This is not needed.

JamesVaughan commented on aimsunBridge.py:
```diff
@@ -218,6 +240,13 @@ def run(self):
+                    self.switchModel()
+                    # send to the pipe that we ran the message successfully
+                    self.sendSuccess()
+                    self.sendSignal(self.SignalStart)
```
We don't need to send another Start signal.  The success message is going to be enough to let XTMF or the unit tests know that it can continue processing.  Additionally the success signal should be sent in the switchModel method.

amitsandhel committed:
Re-factored the code based on the Github feedback.
Now the switchModel() does run and close successfully as intended.

amitsandhel committed:
Re-factored the json parser
Passing parameters more flexibly.
Still need to clean up the codebase a bit more

amitsandhel committed:
Removed dead code on line 515 in importnetwork.py

amitsandhel committed:
removed the aimsuncontroller debuggerattached commented code as we don't need it anymore
changed SwitchNetworkPath to SignalSwitchNetworkPath to follow appropaite naming convention
added comments to the TestConfiguration class

amitsandhel committed:
added space between methods in aimsuncontroller.cs
testconfiguration helper class comments are triple slashed for better documentation
inlined the json creation data
removed second unittest as it was redundant and not needed

amitsandhel committed:
fixed the grammar typos and removed the launchaimsun input variable as it is not needed anymore

JamesVaughan reviewed (approved):
LGTM
